map.go

```diff
@@ -972,9 +973,13 @@ func (m *Map) unmarshalValue(value interface{}, buf []byte) error {
 	return unmarshalBytes(value, buf)
 }
 
-// LoadPinnedMap load a Map from a BPF file.
-func LoadPinnedMap(fileName string) (*Map, error) {
-	fd, err := internal.BPFObjGet(fileName)
+// LoadPinnedMap loads a Map from a BPF file.
+func LoadPinnedMap(fileName string, opts *LoadPinOptions) (*Map, error) {
```

Comment:
AFAICS, the 2nd argument doesn't need to be a pointer. You can then get rid of the below `nil` check.

Reply:
The idea is that most callers won't want to set flags. With the current API that looks like:
```
LoadPinnedMap(path, nil)
```
Without a pointer we need this everywhere:
```
LoadPinnedMap(path, ebpf.LoadPinOptions{})
```

That's a lot more cumbersome for the common case.

The current approach of the library is to have two functions: `LoadPinnedMap()` and `LoadPinnedMapWithOptions(string, struct{})`. The upside is no breaking changes, the downside a lot of API bloat. After discussing with @ti-mo I think a single function with "optional" options via a pointer struct is the way forward.

Reply:
I don't think this is a good trade off, where users are now expected to know that they need to pass nil as the 2nd argument. What's the more common case?

Reply:
For better or worse we already have APIs like this. Map.Batch* or CollectionSpec.LoadAndAssign already have an optional pointer, while other parts of the library have `WithOptions`, while creating a link takes a non-pointer struct. Right now there is no consistency, which I would like to fix. This API seems like a step in the right direction, but let me mull it over some more.

Can you explain more why you don't like the `nil` argument? Why is documenting that options can be nil not sufficient?

Maybe @ti-mo can weigh in as well, he seems to like the `*struct{}` style.

Reply:
It's not so much about whether the argument is `nil` or not. It's mainly about user experience, where I feel we are forcing them to pass an additional argument that they don't need in the most common case.

Reply:
The `nil` config struct idea comes from https://github.com/mdlayher/netlink/blob/master/conn.go#L57, which has worked very well to keep the amount of exported functions down in packages with significant API surface. (like the netlink family of packages)

- Single, unambiguous way of getting something done (`FooBar()` vs. `FooBarWithOptions()`, why let the user make a choice?)
- Callers that want default behaviour pass `nil`
- Relatively painless for users to migrate if (when) we break API (`FooBar("x")` -> `FooBar("x", nil)`, `FooBarWithOptions("x", FooBarOptions{"y"})` -> `FooBar("x", &FooBarOptions{"y"})`
- Config struct can be extended without breaking API (:rocket:)
- It keeps godoc tidy and terse

Personally, I've always been used to this style, so `FooBarWithOptions()` looks strange (and more like Java) to me.

Reply:
Ok, thanks for the context @ti-mo. I still have a preference for adding a new API for the some of the reasons that you mentioned. 🙃 @lmb As the author of the PR, I'll leave the final decision to you. Marking this as resolved.